File: java/src/test/java/com/alxarafe/domain/topology/service/LocationCodeGeneratorTest.java
package com.alxarafe.domain.topology.service;

import com.alxarafe.app.domain.topology.service.LocationCodeGenerator;
import com.alxarafe.app.domain.topology.valueobject.NamingPolicy;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class LocationCodeGeneratorTest {

    private LocationCodeGenerator generator;

    @BeforeEach
    void setUp() {
        generator = new LocationCodeGenerator();
    }

    @Test
    void generateWithDashSeparator() {
        var policy = new NamingPolicy("-", 2, 2, 2, 3, 2);
        var code = generator.generate(policy, "WH", "Z", "A", 1, 1);
        assertEquals("WH-0Z-0A-001-01", code.value());
    }

    @Test
    void generateWithDotSeparator() {
        var policy = new NamingPolicy(".", 3, 2, 3, 2, 1);
        var code = generator.generate(policy, "WH", "Z", "A", 1, 1);
        assertEquals("0WH.0Z.00A.01.1", code.value());
    }

    @Test
    void generateWithLargeNumbers() {
        var policy = new NamingPolicy("-", 2, 2, 2, 3, 2);
        var code = generator.generate(policy, "WH", "Z", "A", 999, 99);
        assertEquals("WH-0Z-0A-999-99", code.value());
    }

    @Test
    void generateWithNoSeparator() {
        var policy = new NamingPolicy("_", 1, 1, 1, 1, 1);
        var code = generator.generate(policy, "W", "Z", "A", 1, 1);
        assertEquals("W_Z_A_1_1", code.value());
    }
}

File: php/tests/Unit/Domain/Topology/Service/LocationCodeGeneratorTest.php
<?php

declare(strict_types=1);

namespace Tests\Unit\Domain\Topology\Service;

use Alxarafe\App\Domain\Topology\Service\LocationCodeGenerator;
use Alxarafe\App\Domain\Topology\ValueObject\NamingPolicy;
use PHPUnit\Framework\TestCase;

final class LocationCodeGeneratorTest extends TestCase
{
    private LocationCodeGenerator $generator;

    protected function setUp(): void
    {
        $this->generator = new LocationCodeGenerator();
    }

    public function testGenerateWithDashSeparator(): void
    {
        $policy = new NamingPolicy('-', 2, 2, 2, 3, 2);
        $code = $this->generator->generate($policy, 'WH', 'Z', 'A', 1, 1);
        self::assertSame('WH-0Z-0A-001-01', $code->value());
    }

    public function testGenerateWithDotSeparator(): void
    {
        $policy = new NamingPolicy('.', 3, 2, 3, 2, 1);
        $code = $this->generator->generate($policy, 'WH', 'Z', 'A', 1, 1);
        self::assertSame('0WH.0Z.00A.01.1', $code->value());
    }

    public function testGenerateWithLargeNumbers(): void
    {
        $policy = new NamingPolicy('-', 2, 2, 2, 3, 2);
        $code = $this->generator->generate($policy, 'WH', 'Z', 'A', 999, 99);
        self::assertSame('WH-0Z-0A-999-99', $code->value());
    }

    public function testGenerateWithNoSeparator(): void
    {
        $policy = new NamingPolicy('_', 1, 1, 1, 1, 1);
        $code = $this->generator->generate($policy, 'W', 'Z', 'A', 1, 1);
        self::assertSame('W_Z_A_1_1', $code->value());
    }
}
